analysis/tools/ccsh/src/main/kotlin/de/maibornwolff/codecharta/tools/ccsh/parser/ParserService.kt

```diff
@@ -1,55 +1,83 @@
+import de.maibornwolff.codecharta.tools.ccsh.parser.repository.PicocliParserRepository
 import de.maibornwolff.codecharta.tools.interactiveparser.InteractiveParser
 import picocli.CommandLine
 
 class ParserService {
     companion object {
         private const val EXIT_CODE_PARSER_NOT_SUPPORTED = 42
 
-        fun selectParser(commandLine: CommandLine): String {
+        fun offerParserSuggestions(commandLine: CommandLine, parserRepository: PicocliParserRepository, inputFile: String): List<String> {
+            val allParsers = parserRepository.getAllParsers(commandLine)
+            val usableParsers = parserRepository.getUsableParserNames(inputFile, allParsers)
+
+            return if (usableParsers.isEmpty()) {
+                emptyList()
+            } else {
+                KInquirer.promptCheckbox(
+                        message = "Choose from this list of applicable parsers",
+                        choices = usableParsers)
+            }
+        }
+
+        fun configureParserSelection(commandLine: CommandLine, parserRepository: PicocliParserRepository, selectedParsers: List<String>): Map<String, List<String>> {
+            val configuredParsers = mutableMapOf<String, List<String>>()
+            for (selectedParser in selectedParsers) {
+                val interactiveParser = parserRepository.getParser(commandLine, selectedParser)
+                if (interactiveParser == null) {
+                    throw IllegalArgumentException("Tried to configure non existing parser!")
+                } else {
+                    val configuration = interactiveParser.getDialog().collectParserArgs()
+                    configuredParsers[selectedParser] = configuration
+
+                    println("You can run the same command again by using the following command line arguments:")
+                    println("ccsh " + selectedParser + " " + configuration.map { x -> '"' + x + '"' }.joinToString(" "))
+                }
+            }
+            return configuredParsers
+        }
+
+        fun selectParser(commandLine: CommandLine, parserRepository: PicocliParserRepository): String {
             val selectedParser: String = KInquirer.promptList(
                 message = "Which parser do you want to execute?",
-                choices = getParserNamesWithDescription(commandLine)
+                choices = parserRepository.getParserNamesWithDescription(commandLine)
             )
-
-            return extractParserName(selectedParser)
+            return parserRepository.extractParserName(selectedParser)
         }
 
         fun executeSelectedParser(commandLine: CommandLine, selectedParser: String): Int {
             val subCommand = commandLine.subcommands.getValue(selectedParser)
             val parserObject = subCommand.commandSpec.userObject()
             val interactive = parserObject as? InteractiveParser
-            if (interactive != null) {
+            return if (interactive != null) {
```

Comment:
The function has Int as return type. Why did you remove the return statement? :-)

Reply:
> I moved the return in front of the if - should this not just return the old value or did I understand this wrong?
Same reasoning,  I just moved the return in front of the if - I think I will revert that change though, since it is a bit confusing.